gadget/validate_test.go

```diff
@@ -298,18 +353,91 @@ volumes:
 	// test error with no volumes
 	makeSizedFile(c, filepath.Join(s.dir, "meta/gadget.yaml"), 0, []byte(bloader))
 
-	constraints := &modelConstraints{
+	mod := &modelCharateristics{
 		systemSeed: true,
 	}
-	ginfo, err := gadget.ReadInfo(s.dir, constraints)
+
+	ginfo, err := gadget.ReadInfo(s.dir, mod)
+	c.Assert(err, IsNil)
+	err = gadget.Validate(ginfo, mod, nil)
+	c.Assert(err, ErrorMatches, "model requires system-seed partition, but no system-seed or system-data partition found")
+}
+
+func (s *validateGadgetTestSuite) TestValidateSystemRoleSplitAcrossVolumes(c *C) {
```

Comment:
it's fine if you don't want to add it to this pr, but I think we should add a test case with all of the ubuntu-* partitions/roles on one volume and another volume with unrelated partitions on it to demonstrate that multiple volumes works with uc20, as it's my understanding that this pr would enable that setup

Reply:
yea, this seems follow up material